### PR TITLE
Added CVSS-related tests (6.1.8, 6.1.9, 6.1.10)

### PR DESCRIPTION
**Pull request by oxisto**
This PR adds the mandatory tests for CSAF document conformance related to the CVSS standard. Since no CVSS standard implementation is available in pure Kotlin, we decided to implement CVSS v2 and v3 in a separate module `csaf-cvss` and employ our solution in the validation tests.

Depends on #67

### CODE REVIEW

File: csaf-cvss/src/main/kotlin/io/github/csaf/sbom/cvss/v3/Calculation.kt [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+fun CvssV3Calculation.roundUp(x: Double): Double {
+    return if (version == "3.1") {
+        var intInput = (x * 100000).roundToInt()
+        if ((intInput % 10000) == 0) {
+            intInput / 100000.0
+        } else {
+            return (floor(intInput / 10000.0) + 1) / 10.0
+        }
+    } else {
+        val factor = 10.0.pow(1)
```
Comment (milux) [inline csaf-cvss/src/main/kotlin/io/github/csaf/sbom/cvss/v3/Calculation.kt:406]: Should be simplified.

File: csaf-cvss/src/main/kotlin/io/github/csaf/sbom/cvss/v3/Calculation.kt [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+ * For CVSS 3.1, it follows the implementation guidance in
+ * [Appendix A](https://www.first.org/cvss/v3.1/specification-document#Appendix-A---Floating-Point-Rounding).
+ */
+fun CvssV3Calculation.roundUp(x: Double): Double {
+    return if (version == "3.1") {
+        var intInput = (x * 100000).roundToInt()
+        if ((intInput % 10000) == 0) {
+            intInput / 100000.0
+        } else {
+            return (floor(intInput / 10000.0) + 1) / 10.0
```
Comment (milux) [inline csaf-cvss/src/main/kotlin/io/github/csaf/sbom/cvss/v3/Calculation.kt:403]: `return` can be removed.

File: csaf-validation/src/test/kotlin/io/github/csaf/sbom/validation/tests/TestsTest.kt
Comment (milux) [inline csaf-validation/src/test/kotlin/io/github/csaf/sbom/validation/tests/TestsTest.kt:1]: `csaf` directory seems empty. Missing test files?

**Review by milux — CHANGES_REQUESTED** [review]

File: csaf-validation/src/test/kotlin/io/github/csaf/sbom/validation/tests/TestsTest.kt
Comment (oxisto) [reply csaf-validation/src/test/kotlin/io/github/csaf/sbom/validation/tests/TestsTest.kt:1]: git submodule :)

File: csaf-cvss/src/main/kotlin/io/github/csaf/sbom/cvss/Extensions.kt [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    val metrics = mutableMapOf<String, String>()
+
+    for ((idx, part) in parts.withIndex()) {
+        val keyValue = part.split(":")
+        val key = keyValue.first()
+        val value = keyValue.getOrNull(1)
+
+        if (allowedVersions != null) {
+            if (idx == 0 && (key != "CVSS" || (value !in allowedVersions))) {
+                // First key must be CVSS:3.X
```
[the remark this answers is not in the document]
Comment (oxisto) [reply csaf-cvss/src/main/kotlin/io/github/csaf/sbom/cvss/Extensions.kt:58]: This is tricky, maybe needs to documented better. It seems that the CVSS 2.0 vector string does NOT have the CVSS prefix, only the later ones have it. In the CVSS 2.0 case the `allowedVersions` needs to be set to null.

**Review by milux — APPROVED** [review]

**Review by milux — APPROVED** [review]

**Pull request merged**